dashboard/app/models/gallery_activity.rb

```diff
@@ -26,12 +26,18 @@ class GalleryActivity < ActiveRecord::Base
   belongs_to :user
   belongs_to :activity
   belongs_to :level_source
+  belongs_to :user_level
 
   before_save :set_app
 
   def set_app
-    return unless activity
-    self.app = activity.try(:level).try(:game).try(:app)
+    if user_level
+      self.app = user_level.level.try(:game).try(:app)
+      return
+    end
```

Comment:
how about 
if user_level
self.app = ...
else if level_source
self.app = ...
end

Then you can skip the return on 36